challenge.go

```diff
@@ -2,43 +2,115 @@ package srp
 
 import (
 	"bytes"
+	"crypto/hmac"
 	"crypto/sha256"
 	"encoding/base64"
 	"encoding/binary"
 	"errors"
+	"math"
+	"time"
 
+	"golang.org/x/crypto/argon2"
 	"golang.org/x/crypto/curve25519"
 )
 
-const ECDLPEphemeralSize = 24
+// Implementation following the "context" package
+var DeadlineExceeded error = deadlineExceededError{}
 
-func ECDLPChallenge(b64Challenge string) (solution int64, err error) {
-	if len(b64Challenge) != 76 { // 56 bytes in base64
-		return 0, errors.New("srp:invalid ECDLP challenge length")
+type deadlineExceededError struct{}
+
+func (deadlineExceededError) Error() string   { return "srp: deadline exceeded calculating proof-of-work challenge" }
+func (deadlineExceededError) Timeout() bool   { return true }
+func (deadlineExceededError) Temporary() bool { return true }
+
+const ecdlpPRFKeySize = 32
+
+// Deadlines are measured on the wall clock, not the monotonic clock, due
+// to unreliability on mobile devices
```

Comment:
Top-level comments should be in the form `functionName` .... and terminated by a dot
```suggestion
// ECDLPChallenge computes the base64 solution for a given ECDLP base64 challenge 
// within deadlineUnixMilli milliseconds, if any was found. Deadlines are measured on the
// wall clock, not the monotonic clock, due to unreliability on mobile devices.
```